mono/utils/mono-os-mutex.h

```diff
@@ -191,6 +188,22 @@ mono_os_cond_timedwait (mono_cond_t *cond, mono_mutex_t *mutex, guint32 timeout_
 
 	/* ms = 10^-3, us = 10^-6, ns = 10^-9 */
 
+#ifdef HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP
+	memset (&ts, 0, sizeof (struct timespec));
+	ts.tv_sec = timeout_ms / 1000;
+	ts.tv_nsec = (timeout_ms % 1000) * 1000 * 1000;
+
+	res = pthread_cond_timedwait_relative_np (cond, mutex, &ts);
+	if (G_UNLIKELY (res != 0 && res != ETIMEDOUT)) {
+		g_print ("cond: %p mutex: %p\n", *(gpointer*)cond, *(gpointer*)mutex);
+		g_error ("%s: pthread_cond_timedwait_relative_np failed with \"%s\" (%d) %ld %ld %d", __func__, g_strerror (res), res, ts.tv_sec, ts.tv_nsec, timeout_ms);
+	}
+	return res != 0 ? -1 : 0;
+#else
+#ifdef BROKEN_CLOCK_SOURCE
```

Comment:
I mean the one for `BROKEN_CLOCK_SOURCE`, not `HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP`.